sky/backends/backend_utils.py

```diff
@@ -1355,8 +1384,8 @@ def check_network_connection():
     try:
         http.head(_TEST_IP, timeout=3)
     except requests.Timeout as e:
-        raise exceptions.NetworkError(
-            'Could not refresh the cluster. Network seems down.') from e
+        raise exceptions.NetworkError('Network seems down.') from e
+    check_if_ssh_blocked()
```

Comment:
Suggest removing the `try...except` clause if `check_if_ssh_blocked` exists, since the method also checks for network connection.